src/object-to-ast.ts

```diff
@@ -0,0 +1,46 @@
+import type babelCore from '@babel/core'
+
+type ObjectToASTOptions = {
+  babel: typeof babelCore
+  fileOptions: babelCore.TransformOptions
+}
+
+export default function objectToAST(
+  object: unknown,
+  {babel, fileOptions}: ObjectToASTOptions,
+) {
+  const stringified = stringify(object)
+  const variableDeclarationNode = babel.template(`var x = ${stringified}`, {
+    preserveComments: true,
+    placeholderPattern: false,
+    ...fileOptions.parserOpts,
+    sourceType: 'module',
+  })() as babelCore.types.VariableDeclaration
+
+  return variableDeclarationNode.declarations[0].init
+}
+
+function stringify(object: unknown) {
+  let str = JSON.stringify(object, stringifyReplacer)
+
+  // This check is necessary as `JSON.stringify` can return undefined,
+  // however the TypeScript definition for it does not include it.
+  // See https://github.com/microsoft/TypeScript/issues/18879 for details.
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  if (str === undefined) {
+    str = 'undefined'
+  }
```

Comment:
this can be shortened and please ts if we do something like this
```ts
const str = JSON.stringify(object, stringifyReplacer) ?? 'undefined'
```

Reply:
nice.  I like it!

Note, the warning still persists, so the comment and disable line need to stay 😭